core/chains/evm/logpoller/log_poller.go

```diff
@@ -558,22 +564,16 @@ func (lp *logPoller) BackupPollAndSaveLogs(ctx context.Context, backupPollerBloc
 			}
 			return
 		}
-
+		finalityDepth := latestBlock.Number - lastFinalizedBlockNumber
```

Comment:
I think this is incorrect. Say we crash with finality depth 10, then reboot and latest finality depth is 5. Ignoring backup delay, we'll set backup poll to 5 when it should be 10. So we'll miss backup polling 5 blocks. Simple solution : use lastProcessed.LastFinalizedBlock number instead. I think whole thing can be simplified:
```
lp.backupPollerNextBlock = mathutil.Min(lastProcessed.LastFinalizedBlock + 1, lastProcessed.Number - backupPollerBlockDelay)
```
Can move the latest query back to where it was that way as well.

Reply:
Wow, good catch!

It looks like the existing logic is equivalent to `mathutil.Min(lastProcessed.LastFinalizedBlock - 1, lastProcessed.Number - backupPollerBlockDelay)` though.  Was it off by 2 blocks? Instead of starting at the block before last finalized, now we'd be using the block after the last finalized.

I think that sounds right, just wanted to confirm that this is an intentional fix of another case where the production code was "more safe than necessary", rather than keeping functionality the same and just refactoring.

Reply:
Agree, you are right @connorwstein. Btw, we should have more tests that verify edge cases like these, I think the problem is also present in the current `develop`, as we subtract `finalityDepth` from the current tip of the blockchain

Reply:
you are right @reductionista. Let's keep logic as is with starting from finalized - 1. The benefit is just consistency: everywhere we backfill, we backfill up to finalized - 1 (we need the finalized block itself to be able to reorg check on finalized+1).